controllers/openstackdataplanenode_controller.go

```diff
@@ -303,10 +303,12 @@ func (r *OpenStackDataPlaneNodeReconciler) GetInstanceRole(ctx context.Context,
 	// Use the instances's role name to get its role object
 	var err error
 	instanceRole := &dataplanev1beta1.OpenStackDataPlaneRole{}
-	err = r.Client.Get(ctx, client.ObjectKey{
-		Namespace: instance.Namespace,
-		Name:      instance.Spec.Role,
-	}, instanceRole)
+	if len(instance.Spec.Role) > 0 {
+		err = r.Client.Get(ctx, client.ObjectKey{
+			Namespace: instance.Namespace,
+			Name:      instance.Spec.Role,
+		}, instanceRole)
+	}
```

Comment:
I agree about only running Get if there was a role set. Regardless of if we want to make the role mandatory or optional, either way we should have this code (306-311).  If we go optional, then we could have something like this.

  https://github.com/openstack-k8s-operators/dataplane-operator/pull/69

Reply:
+1